File: dashboard/backend/models/Report.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class Report
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public required string UserId { get; set; }
    public int CategoryId { get; set; }
    public DateTime? CreateDate { get; set; }
    public DateTime? ReportDate { get; set; }
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }
    public required List<string> Images { get; set; }
    public required string Description { get; set; }
    public int? InvolvedActorId { get; set; }
    public int? VictimActorId { get; set; }

    [ForeignKey("CategoryId")]
    public virtual required Category Category { get; set; }

    [ForeignKey("InvolvedActorId")]
    public virtual VialActor? InvolvedActor { get; set; }

    [ForeignKey("VictimActorId")]
    public virtual VialActor? VictimActor { get; set; }

}

File: dashboard/omus/lib/services/models/report.dart
class Report {
  final int id;
  final String? userId;
  final int categoryId;
  final DateTime? createDate;
  final DateTime? reportDate;
  final double? latitude;
  final double? longitude;
  final List<String>? images;
  final String? description;
  final int? involvedActorId;
  final int? victimActorId;

  Report({
    required this.id,
    this.userId,
    required this.categoryId,
    this.createDate,
    this.reportDate,
    this.latitude,
    this.longitude,
    this.images,
    this.description,
    this.involvedActorId,
    this.victimActorId,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['id'],
      userId: json['userId'],
      categoryId: json['categoryId'],
      createDate: DateTime.tryParse(json['createDate'] ?? ""),
      reportDate: DateTime.tryParse(json['reportDate'] ?? ""),
      latitude: json['latitude'] != null
          ? (json['latitude'] as num).toDouble()
          : null,
      longitude: json['longitude'] != null
          ? (json['longitude'] as num).toDouble()
          : null,
      images: (json['images'] as List?)?.map((e) => e as String).toList(),
      description: json['description'],
      involvedActorId: json['involvedActorId'],
      victimActorId: json['victimActorId'],
    );
  }
}
